Provider/src/main/java/com/spotify/confidence/ConfidenceValue.kt

```diff
@@ -0,0 +1,26 @@
+package com.spotify.confidence
+
+import com.spotify.confidence.client.serializers.ConfidenceValueSerializer
+import dev.openfeature.sdk.DateSerializer
+import kotlinx.serialization.Serializable
+
+@Serializable(ConfidenceValueSerializer::class)
+sealed interface ConfidenceValue {
+    @Serializable
+    data class String(val string: kotlin.String) : ConfidenceValue
+
+    @Serializable
+    data class Double(val double: kotlin.Double) : ConfidenceValue
+
+    @Serializable data class Boolean(val boolean: kotlin.Boolean) : ConfidenceValue
+
+    @Serializable data class Integer(val integer: Int) : ConfidenceValue
+
+    @Serializable data class Struct(val map: Map<kotlin.String, ConfidenceValue>) : ConfidenceValue
+
+    @Serializable data class List(val list: kotlin.collections.List<ConfidenceValue>) : ConfidenceValue
+
+    @Serializable data class Date(@Serializable(DateSerializer::class) val date: java.util.Date) : ConfidenceValue
```

Comment:
I'm fairly certain that this format (defined by this serializer) is not what is expected for a Date type in an event schema. Did you try this against the backend?

Reply:
is the Date type in event schema different than flag type for date? or evaluation context? because they should be the same, since we are using the same types. if they are different backend should change either for flags or events to support one single format.